### ISSUE TITLE
Plex Only Recording Partial Episodes

### ISSUE THREAD

**Issue by sacoward** [tgorgdotcom/locast2plex#258]
Recently I began to have an issue where Plex is only recording partial episodes via locast2plex.  The recordings are anything from a couple of seconds to 5 or 6 minutes.  I have tried rebooting my raspi and restarting locast2plex . . . as well as restarting my plex server.  Any ideas?  Thanks!

**Comment by rocky4546** [comment]
The simple answer is that locast2plex is playing only the first MPEGTS file that is 6 seconds long for HD and 10 seconds long for SD.  Following that, it is terminating the stream.  It is rare for Locast to not provide a continuous supply of MPEGTS files to stream, so not sure why the issue.  You can try to play the stream from VLC.  Installing VLC on your normal desktop and then pointing to the http://hostname:6077/channels.m3u should give you a channel list in VLC to try out and see if the same issue is happening.  That will tell you if plex is the cause.

**Comment by sacoward** [comment]
Would this be the case for those times where it records for 3, 4, or 5 minutes?

**Comment by rocky4546** [comment]
I use tvheadend instead of plex and it is possible that plex is terminating the stream if a long enough delay occurs between stream files.  So you could get to 5 minutes in and then a delay occurs in getting the next file from locast causing the timeout.  May thing is to try it on VLC.  What VLC will do when a delay occurs is to start playing the next channel.  So, let it run and when you come back, if it is playing another channel, then you probably have delays occurring.  If that is the case, I have some possible solutions to the issue.

**Comment by sacoward** [comment]
In VLC I never get anything when I try to add the above link (I do replace hostname with the IP address of the computer.  When I try to "open the location from clipboard"  . . . the terminal window of locast I get the follow "192.168.1.3 - - [12/May/2021 14:30:57] "GET /channels.m3u HTTP/1.1" 200 -"

**Comment by rocky4546** [comment]
The log line implies you have an application helper setup in your browser and it is automatically sending it elsewhere to an app that cannot handle it.  Based on the browser, you need to find a way to save the file to a folder.  Then you can open it with VLC. An example is Firefox under Options > General has an Application section where M3U files and others are tagged for processing.  changing it to Always Ask will allow you to save the file or push it to VLC.

**Comment by sacoward** [comment]
Thank you.  I plugged the url into Brave and it asked me if I wanted to download it.  I opened it up in VLC without an issue.  Currently VLC is playing now for 8 minutes without changing to the next channel.  It has paused for like 1 second  . . . two or three times.

**Comment by rocky4546** [comment]
The one second delays are caused by initial buffering.  The default read_buffer for locast2plex is large (1152000 bytes) and that can take around 6 seconds to fill before it sends the buffer onward.  It really needs about 6-8 seconds of buffering before it will not pause.  So a few times of pausing is normal.  When it is to the DVR, buffer it not an issue since it just takes the stream as it gets it (within reason).

I did think of another possible issue a few had about a month ago.  Some of the weaker channels would not play well.  The stream would either play the video without sound or it would have snapshots with the sound playing.  The stronger channels did not have this issue.  It was tracked to Locast and they even had the same issue on their web site.

**Comment by sacoward** [comment]
So I tried some recording on what I would assume would be stronger channels (ABC, NBC, CBS) last night all with the same results.  At this point would it be best to try to start running locast2plex on the same windows computer I run plex on via docker (currently it is running on a raspi)?

**Comment by rocky4546** [comment]
I have a thought. Can you contact me on Discord at rocky4546#8998

**Comment by joerich** [comment]
Great way to help other people with the same problem!
take the conversation somewhere else?????
You suck Rocky!

**Comment by sacoward** [comment]
The issue is still occurring.  I have also tried an uninstall/reinstall of plex, but that didn't fix anything (although I probably need to try again and remove all traces of plex, so it is a "fresh" install).

**Comment by rocky4546** [comment]
I guess we need to give a summary of what we did.  It was too much back and forth to put here, but a summary works for me.  We actually tried to move to different re-stream apps and he re-installed plex.  Nothing seem to fix the issue.  We could have tried moving to something besides plex which would be jellyfin, but he needs plex to work. It looks like the issue is either his infrastructure or the computer.  I requested he crank up logging and see if something occurs at that level and report that.

**Comment by CTJohnK** [comment]
I've noticed an issue lately of recordings appearing in the DVR Schedule/Recording Priority which noted "No upcoming airings".  Doing a bit more digging, these typically are movies and I have "Allow Partial Recordings" unchecked.

A bit more digging, I noticed these movies are 3-hours long and I have Plex starting a minute early and ending a minute late so exceeding the 3-hour Locast connection time which then closes out the file and per the setting, removes the partial recording.

At least that is what it seems to be doing.  I haven't had any issues with TV shows as they run for 30 to 60 minutes.

I also noticed an issue (I currently have two Locast instances - one on a Synology NAS and one on a Raspberry Pi) with the Pi not always completing a TV episode (I have the partial recording checked for TV shows), but that seems to be more an issue with the Pi and perhaps it not being fast enough to run Plex and record more than one show.

**Comment by rocky4546** [comment]
One issue with Plex is that it tends to trans-encode on record.  Transencoding can take a significant amount of CPU cycles if the processor does not have a hardware-based encoder (with software that can use it).  VAAPI is an example.  Not to solo out Plex, many other dvrs also do that.  I personally use them as a media server and DVR with TVHeadend.  By default it does no trans-encoding and uses very little CPU and memory.  If a TVHeadend plugin is available from Plex, I would go in that direction and have your RPi run the TVHeadend and Locast services.  I use plugins on Kodi, Emby, and JellyFin to interface into tvheadend and I have recorded up to 8 channels at one time with 25% CPU usage on an Intel i3. Just a thought.

**Comment by CTJohnK** [comment]
Agreed.  I use the RPi as a second tuner most of the time with my NAS running Plex providing the recording.  So, the RPi doesn't have to do the recording.

**Comment by rub1xcub3** [comment]
I have having a similar issue. All recordings just cut off usually under an hour. 

It doesn't happen while streaming only when recording is happening.

Also, when it runs into it's issue, all channels stop working. I then have to restart the locast2plex script to start working again.
